src/cloud-functions.ts

```diff
@@ -276,19 +263,44 @@ export function makeCloudFunction<EventData>({
     }
     context.params = context.params || _makeParams(context, triggerResource);
 
-    try {
-      before(event);
+    before(event);
 
-      let dataOrChange = dataConstructor(event);
-      let promise = handler(dataOrChange, context);
-      if (typeof promise === 'undefined') {
-        console.warn('Function returned undefined, expected Promise or value');
-      }
-      return await promise;
-    } finally {
-      after(event);
+    let dataOrChange = dataConstructor(event);
+    console.log('right before calling the handler, event is: ', event)
+    let promise = handler(dataOrChange, context);
+    if (typeof promise === 'undefined') {
+      console.warn('Function returned undefined, expected Promise or value');
     }
+    return Promise.resolve(promise)
+      .then(result => {
+        after(event);
+        return result;
+      })
+      .catch(err => {
+        after(event);
+        return Promise.reject(err);
+      });
   };
+
+  if (process.env.NEW_FUNCTION_SIGNATURE) {
+    console.log('in if loop')
+    cloudFunction = cloudFunctionNewSignature;
```

Comment:
Since this is a single line, how about we remove the `if/else` in favor of:

```js
let cloudFunction = cloudFunctionNewSignature = ...;

// wrap in a single-argument function for legacy signature
if (!process.env.NEW_FUNCTION_SIGNATURE === 'true') {
  cloudFunction = (raw) => {
    ...
    return cloudFunctionNewSignature(data, context);
  }
}
```